modules/core/src/main/scala/org/tesselation/infrastructure/trust/TrustDaemon.scala

```diff
@@ -0,0 +1,74 @@
+package org.tesselation.infrastructure.trust
+
+import cats.Parallel
+import cats.effect.std.Random
+import cats.effect.{Async, Spawn, Temporal}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tesselation.config.types.TrustDaemonConfig
+import org.tesselation.domain.Daemon
+import org.tesselation.domain.trust.storage.TrustStorage
+import org.tesselation.kryo.KryoSerializer
+import org.tesselation.schema.peer.PeerId
+import org.tesselation.security.SecurityProvider
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+trait TrustDaemon[F[_]] extends Daemon[F]
+
+object TrustDaemon {
+
+  def make[F[_]: Async: SecurityProvider: KryoSerializer: Random: Parallel](
+    cfg: TrustDaemonConfig,
+    trustStorage: TrustStorage[F],
+    selfPeerId: PeerId
+  ): TrustDaemon[F] = new TrustDaemon[F] {
+
+    private val logger = Slf4jLogger.getLogger[F]
+
+    def start: F[Unit] =
+      for {
+        _ <- Spawn[F].start(modelUpdate.foreverM).void
+      } yield ()
+
+    private def calculateTrust(): F[Unit] = {
```

Comment:
That method breaks the single responsibility principle because it not only calculates trust but also updates the trust.

My suggestion:

```scala
// Keep that one pure so we can unit test that. Also rename it so we know which trust it calculates.
private def calculatePredictedTrust(trustInfo: Map[PeerId, TrustInfo]): Map[PeerId, Double] { ... }

private def modelUpdate: F[Unit] = 
  for {
     _ <- Temporal[F].sleep(cfg.interval)
     predictedTrust <- trustStorage.getTrust.map(calculatePredictedTrust)
     _ <- trustStorage.updatePredictedTrust(predictedTrust)
  } yield ()
```

Reply:
Fixed. Going to start adding tests for this in next PRs as well.